vdyp-lib/vdyp-vri/src/main/java/ca/bc/gov/nrs/vdyp/vri/VriStart.java

```diff
@@ -631,11 +649,143 @@ float quadMeanDiameterSpeciesAdjust(float x, float initialDq, float min, float m
 		@SuppressWarnings("unused")
 		var veteranLayer = polygon.getLayers().get(LayerType.VETERAN);
+		lBuilder.layerType(LayerType.VETERAN);
+
+		lBuilder.adapt(veteranLayer);
+		
+		var bec = polygon.getBiogeoclimaticZone();
+
+		var lowDq = false;
+
+		var primarySite = veteranLayer.getPrimarySite();
+		var primarySpecies = veteranLayer.getPrimarySpeciesRecord();
+
+		float ageTotal = primarySite.flatMap(VriSite::getAgeTotal).map(at -> at + veteranLayer.getAgeIncrease())
+				.orElse(0f); // AGETOTLV
+		float yearsToBreastHeight = primarySite.flatMap(VriSite::getYearsToBreastHeight).orElse(0f); // YTBHLV
+		float breastHeightAge = ageTotal - yearsToBreastHeight; // AGEBHLV
+		float dominantHeight = primarySite.flatMap(VriSite::getHeight).orElse(0f); // HDLV
+
+		var baseArea = veteranLayer.getBaseArea()
+				.orElseThrow(() -> new StandProcessingException("Expected veteran layer to have a base area")); // BA_V1
+		var treesPerHectare = veteranLayer.getTreesPerHectare()
+				.orElseThrow(() -> new StandProcessingException("Expected veteran layer to have a trees per hectare")); // TPH_V1
+
+		treesPerHectare = enforceMinimumDiameter(baseArea, treesPerHectare);
+
+		float inputTreesPerHectare = treesPerHectare; // TPHInput
+
+		if (baseArea <= 0) {
+			throw new StandProcessingException(
+					MessageFormat.format("Veteran layer base area ({0}) was not positive", baseArea)
+			);
+		}
+		if (treesPerHectare <= 0) {
+			throw new StandProcessingException(
+					MessageFormat.format("Veteran layer trees per hectare ({0}) was not positive", baseArea)
+			);
+		}
+		var quadMeanDiameter = BaseAreaTreeDensityDiameter.quadMeanDiameter(baseArea, treesPerHectare); // DQ(0,0)
+
+		lBuilder.adaptSpecies(veteranLayer, (sBuilder, spec) -> {
+			applyGroups(bec, spec.getGenus(), sBuilder);
+			if (primarySpecies.map(spec::equals).orElse(false)) {
+				sBuilder.addSite(iBuilder -> {
+					iBuilder.adapt(primarySite.get());
+					iBuilder.ageTotal(ageTotal);
+					iBuilder.yearsToBreastHeight(yearsToBreastHeight);
+				});
+			}
+			float specHeight = spec.getSite().flatMap(VriSite::getHeight).filter(x -> x > 0).orElse(dominantHeight);
+			float specBaseArea = spec.getFractionGenus() * baseArea;
+			float specQuadMeanDiameter = max(
+					estimateVeteranQuadMeanDiameter(spec.getGenus(), bec, specHeight), VETERAN_MIN_DQ
+			);
+			float specTreeDensity = BaseAreaTreeDensityDiameter.treesPerHectare(specBaseArea, specQuadMeanDiameter);
+			sBuilder.loreyHeight(specHeight);
+			sBuilder.baseArea(specBaseArea);
+			sBuilder.quadMeanDiameter(specQuadMeanDiameter);
+			sBuilder.treesPerHectare(specTreeDensity);
+		});
+
+		lBuilder.buildChildren();
+		var specList = lBuilder.getSpecies();
+
+		// Really crude reconciliation to the specified TPH (if any)
+		// Could improve with code similar to ROOTV01
+		if (treesPerHectare > 0) {
+			float tphSum = 0;
+			for (var spec : specList) {
+				tphSum += spec.getTreesPerHectareByUtilization().getAll();
```

Comment:
The FORTRAN code reads UC Large, not UC All. But of course they may be known to have the same value (although not to me...)